hs_core/models.py

```diff
@@ -1822,30 +1822,113 @@ def check_relations(self, stop_on_error=False, log_errors=True,
                     print("relation {} {} {} (this does not exist)"
                           .format(self.short_id, r.type, target))
 
+    def fix_irods_user_paths(self, log_actions=True, echo_actions=False, return_actions=False):
+        """
+        Move iRODS user paths to the locations specified in settings
+
+        :param log_actions: whether to log actions to Django log
+        :param echo_actions: whether to print actions on stdout
+        :param return_actions: whether to collect actions in an array and return them.
+
+        This is a temporary fix to the user resources, which are currently stored like
+        federated resources but whose paths are dynamically determined. This function points
+        the paths for user-level resources to where they are stored in the current environment,
+        as specified in hydroshare/local_settings.py.
+
+        * This only does something if the environment is not a production environment.
+        * It is idempotent, in the sense that it can be repeated more than once without problems.
+        * It must be done once whenever the django database is reloaded.
+        * It does not check whether the paths exist afterward. This is done by check_irods_files.
+        """
+        logger = logging.getLogger(__name__)
+        actions = []
+        ecount = 0
+
+        # location of the user files in production
+        defaultpath = getattr(settings, 'HS_USER_ZONE_DEFAULT_PATH',
+                              '/hydroshareuserZone/home/localHydroProxy')
+        # where resource should be found; this is equal to the default path in production
+        userpath = '/' + os.path.join(
+            getattr(settings, 'HS_USER_IRODS_ZONE', 'hydroshareuserZone'),
+            'home',
+            getattr(settings, 'HS_LOCAL_PROXY_USER_IN_FED_ZONE', 'localHydroProxy'))
+
+        msg = "fix_irods_user_paths: user path is {}".format(userpath)
+        if echo_actions:
+            print(msg)
+        if log_actions:
+            logger.info(msg)
+        if return_actions:
+            actions.append(msg)
+
+        # only take action if you find a path that is a default user path and not in production
+        if self.resource_federation_path == defaultpath and userpath != defaultpath:
+            msg = "fix_irods_user_paths: mapping existing user federation path {} to {}"\
+                  .format(self.resource_federation_path, userpath)
+            if echo_actions:
+                print(msg)
+            if log_actions:
+                logger.info(msg)
+            if return_actions:
+                actions.append(msg)
+
+            self.resource_federation_path = userpath
+            self.save()
+            for f in self.files.all():
+                path = f.storage_path
```

Comment:
@alvacouch I just realized storage_path() is a property which does not store the absolute path in the db, so we are fine here. However, fed_resource_file field does store the absolute path which I think can be avoided. If fed_resource_file can be changed to only store relative path, then storage_path() property can just concatenate resource_federation_path with fed_resource_file.name rather than returning fed_resource_file.name directly as is now. I don't think this change requires any code change in django_irods which we want to be generic. What do you think? If you make fed_resource_file to store relative path, these resource file loop will not be needed here which will simplify things a lot.